internal/discoverer/discoverer.go

```diff
@@ -0,0 +1,87 @@
+package discoverer
+
+import (
+	"errors"
+	"fmt"
+	"github.com/api7/apisix-seed/internal/utils"
+	"strconv"
+)
+
+var (
+	queryChecks = [3]string{"event", "entity", "service"}
+	Discoveries = make(map[string]Discover)
+)
+
+type Discover func(disConfig interface{}) Discoverer
+
+// Discoverer defines the component that interact nacos, consul and so on
+type Discoverer interface {
+	Stop()
+	Query(utils.Message) error
+	Watch() chan utils.Message
+}
+
+type Node struct {
+	host   string
+	weight int
+}
+
+type Service struct {
+	name     string
+	nodes    []Node
+	entities []string
+	args     map[string]string
+}
+
+// queryDecode check and extract values from the query message
+func queryDecode(msg utils.Message) ([]string, map[string]string, error) {
+	msgLength := len(msg)
+	if msgLength < 3 {
```

Comment:
Please avoid repeating the magic number 3

Reply:
I have reorganized the query and watch messages about the communication between components.